packages/core/apps/core-apps-server-internal/src/core_app.ts

```diff
@@ -41,10 +41,12 @@ interface CommonRoutesParams {
 export class CoreAppsService {
```

Comment:
(unrelated to this file, just writing as a comment to allow replying on the thread)

Overall, the technical implementation makes sense, and I don't have any strong problem regarding it.

My concerns / questions are more regarding the usages we're planning on doing of this feature. 

We're basically implementing a per-node hot configuration update here. Which can be great. Now, these hot changes are not persisted. Any node rebooting will loose those hot updates, and reboot with the 'initial' configuration.

Which makes me wonder if we're planning any concrete **production** usage of the feature, or if the only planned usage is to update those labels in testing environments. 

Because if that's the latest, then I can't avoid wondering if having that as a core API / endpoint makes sense compared to have it as a plugin that would just be in charge of updating the APM and EBT labels accordingly, without having to plug the feature so low level in core's config service.

OTOH, being able somehow to update dynamically some parts the config more easily than how it's currently done (I'm thinking, idk, about enabling debug logging via this endpoint, for instance) could be extremely powerful. 

Curious to know what you think about this?

Reply:
TL;DR, Yes to all! 😇 

> I can't avoid wondering if having that as a core API / endpoint makes sense compared to have it as a plugin that would just be in charge of updating the APM and EBT labels accordingly

I initially implemented that. The main reason I ended up moving it to core was because the plugin couldn't inject the values to the initially rendered page, so APM and EBT initial events like `page-load` were wrongly labeled. Refer to _Alternative options considered_ in the description.

 > OTOH, being able somehow to update dynamically some parts the config more easily than how it's currently done (I'm thinking, idk, about enabling debug logging via this endpoint, for instance) could be extremely powerful.

++ that'd be a great use case for this feature.

> Curious to know what you think about this?

TBH, I have mixed feelings about this solution, especially for the caveats highlighted in _What's not that good about this solution?_. On the flip side, I think it's the best solution we can provide, given the current architecture.

Reply:
Ok, thanks for your explanation. 

In that case, it could be a good idea to talk about it with the team in one of meeting to see what the others think about it?

Reply:
> In that case, it could be a good idea to talk about it with the team in one of meeting to see what the others think about it?

Yeah! In our last sync, I asked the team to have a look at this PR and the description to figure out if this is something we want to support, or if they can help me come up with alternatives.

I'll bring it up again in our next sync :)